[FEATURE] Add support for translation of news records with SUMM AI

eliashaeussler opened the pull request:
## About this PR

Hi, as communicated with you at the end of 2024, we integrated the translation of news records to easy/plain language with SUMM AI into EXT:mkcontentai. In this PR, we propose this feature as an addition to the existing SUMM AI integration of EXT:mkcontentai.

## Features

### Translation into easy/plain language of news records

For each news record, translation into easy/plain language can be triggered in the recordlist:

![image](https://github.com/user-attachments/assets/a68b5d39-91f4-43b8-87bb-85ed60f452f8)

> [!NOTE]
> Translated records are disabled by default.

The original and translated news records are linked using two new fields in `tx_news_domain_model_news`:

![image](https://github.com/user-attachments/assets/2c0aa2c6-bead-42af-a901-30c4a5215189)

---

### Selection of fields to translate

Title, teaser and bodytext of a news record are translated by default. In addition, the content of related content elements is translated as well. It is required to whitelist the CTypes of content elements to translate in the extension settings module:

![image](https://github.com/user-attachments/assets/b0560f9d-33ad-4181-b877-2d840d6ef8fa)

---

### Additional settings

* **Create translations in test mode**: Create dummy instead of real translations (for testing purposes to avoid consuming SUMM AI API tokens)
* **Show a disclaimer that the text was generated by AI**: Appends a (localizable) disclaimer text to each translation, stating that the translation was generated with the help of AI:
  ![image](https://github.com/user-attachments/assets/ca11adf8-6274-4336-bb7b-b32f1d354884)
* **Set content element UID to append to translated news record**: Allows to append a specific content element to each translation. This can be used, for example, to add a link to a general Easy Language page to each news translation.

---

### Permissions for backend users

The Translation into easy/plain language must be enabled for backend users using the extended backend user permissions:

![image](https://github.com/user-attachments/assets/810a553f-2f1c-4ee8-a130-04cf8684bf45)

## Credits

Most of the work was done by @coderanian, who is no longer working at our company. All thanks are due to him!

eliashaeussler committed:
[FEATURE] Add support for translation of news records with SUMM AI

Co-authored-by: Konstantin Kuklin <coderanian@pm.me>

hannesbochmann commented:
Thank you for your effort. We will take a deeper look into the PR. But I can already say that we will probably not accept it. All this functionality should be placed in a separate extension like mkcontentai_news. We would be happy to add missing events so that mkcontentai can be extended properly. 

In addition, there are some coding issues that would need to be addressed if we should take over the code and maintain it from now on. As a starting point running "composer runqa" should fix several things considering the coding style. Of course the tests should run successful. Adding tests for the new functionality would be more than appreciated as well. After that we will need to see if the CI pipeline (https://github.com/DMKEBUSINESSGMBH/typo3-mkcontentai/actions/runs/14077593760) is fixed. But again it's really uncertain if this PR gets merged after all.

eliashaeussler committed:
[BUGFIX] Assure PHP 7.4 compatibility

eliashaeussler committed:
[TASK] Apply coding style

eliashaeussler committed:
[TASK] Add EXT:news to require-dev section in composer.json

eliashaeussler committed:
Merge branch '12.4' into feature/summ-ai-news-translation

eliashaeussler committed:
[BUGFIX] Allow translating hidden news records

eliashaeussler committed:
[BUGFIX] Include bodytext in news content to translate

eliashaeussler committed:
[TASK] Connect form labels with fields in SUMM AI settings

eliashaeussler committed:
[BUGFIX] Configure `foreign_table` for news columns

eliashaeussler committed:
[TASK] Use IRRE fields to improve usability of news columns

rpflamm committed:
`Handle null bodytext in getBodytext method`

Ensure the `getBodytext` method returns an empty string instead of null when `bodytext` is not set. This prevents potential errors when accessing or processing the method's return value.

rpflamm committed:
Add support for news table and improve TCA field configuration

rpflamm committed:
Refactor Persistence configuration and improve type annotations in News model

eliashaeussler committed:
[FEATURE] Separate translation of original news ans linked news record

konhill committed:
[TASK] Add validation and error handling for input text exceeding 10,000 characters

konhill committed:
[TASK] Define MAX_INPUT_TEXT_LENGTH constant in AiTranslationContentService

konhill committed:
[TASK] Use imported LocalizationUtility explicitly in AiTranslationContentService

konhill committed:
[TASK] Update error messages to specify "Summ AI Translation API" for clarity

konhill committed:
[TASK] Add `ext-mbstring` dependency in composer.json

konhill committed:
[TASK] Reorder  dependency in composer.json

konhill committed:
Merge pull request #2 from CPS-IT/feature/check-api-input-text-lengh

[TASK] Add validation and error handling for input text exceeding 10,000 characters
